packages/cli-repl/src/update-notification-manager.ts

```diff
@@ -95,9 +95,15 @@ export class UpdateNotificationManager {
         : {},
     });
 
+    if (!response.body) {
+      throw new Error(
+        `Missing body from ${updateURL}: ${response.status} ${response.statusText}`
+      );
+    }
+
     if (response.status === 304 /* Not Modified, i.e. ETag matched */) {
       response.body
-        .on('error', () => {
+        .once('error', () => {
```

Comment:
This is probably fine either way – we shouldn't be getting more than one `error` event – but just out of curiosity, did anything in node-fetch change around error handling?

Reply:
yes, https://github.com/node-fetch/node-fetch/blob/main/docs/v3-UPGRADE-GUIDE.md#a-stream-pipeline-is-now-used-to-forward-errors

Reply:
Interesting! Thank you for the link :)